pydoctor/extensions/deprecate.py

```diff
@@ -49,34 +50,38 @@ def getDeprecated(self:model.Documentable, decorators:Sequence[ast.expr]) -> Non
                         section='deprecation text',)
                     self.extra_info.append(parsed_info)
 
-
-class ModuleVisitor(zopeinterface.ZopeInterfaceModuleVisitor):
-    def visit_ClassDef(self, node:ast.ClassDef) -> None:
+class ModuleVisitor(astutils.NodeVisitorExt):
+    visitor: astbuilder.ModuleVistor
+    when = astutils.NodeVisitorExt.When.AFTER
+    
+    def depart_ClassDef(self, node:ast.ClassDef) -> None:
         """
-        Called when a class definition is visited.
+        Called after a class definition is visited.
         """
-        super().visit_ClassDef(node)
         try:
-            cls = self.builder.current.contents[node.name]
+            cls = self.visitor.builder.current.contents[node.name]
         except KeyError:
             # Classes inside functions are ignored.
             return
         assert isinstance(cls, model.Class)
-        getDeprecated(cls, cls.raw_decorators)
+        getDeprecated(cls, node.decorator_list)
 
     def visit_FunctionDef(self, node:ast.FunctionDef) -> None:
         """
-        Called when a function definition is visited.
+        Called after a function definition is visited.
         """
-        super().visit_FunctionDef(node)
         try:
-            func = self.builder.current.contents[node.name]
+            # Property
+            func = self.visitor.builder.current.contents[node.name]
         except KeyError:
-            # Inner functions are ignored.
-            return
+            if self.visitor.builder.current.name == node.name:
+                # Function
+                func = self.visitor.builder.current
+            else:
+                # Inner functions are ignored.
+                return
         assert isinstance(func, (model.Function, model.Attribute))
```

Comment:
Maybe there is better way to get the last function/property visited. I'll try to use the `depart_FunctionDef` instead. 

Also, just realized but this code does not take into account async functions!